generators/csharp/codegen/src/asIs/JsonConfiguration.Template.cs

```diff
@@ -17,6 +18,24 @@ static JsonOptions()
         };
         ConfigureJsonSerializerOptions(options);
         JsonSerializerOptions = options;
+
+        JsonSerializerOptionsWithFallback = new JsonSerializerOptions(options)
+        {
+            TypeInfoResolver = new DefaultJsonTypeInfoResolver
+            {
+                Modifiers =
+                {
+                    static typeInfo =>
+                    {
+                        foreach (var propertyInfo in typeInfo.Properties)
+                        {
+                            // Strip IsRequired constraint from every property.
+                            propertyInfo.IsRequired = false;
+                        }
+                    }
+                }
+            }
+        };
```

Comment:
The `ConfigureJsonSerializerOptions` method is an extension point for SDK maintainers so they can customize some of the options.
I think that this would prevent them from modifying the type info resolvers and modifiers like we are doing.

Reply:
yeah thats fair, i'm not sure how valuable it is to keep that extension method